src/lib/homestore.cpp

```diff
@@ -321,43 +253,46 @@ void HomeStore::init_cache() {
 }
 #endif
 
-void HomeStore::new_vdev_found(DeviceManager* dev_mgr, vdev_info_block* vb) {
+shared< VirtualDev > HomeStore::create_vdev_cb(const vdev_info& vinfo, bool load_existing) {
+    shared< VirtualDev > ret_vdev;
     auto& hs_config = HomeStoreStaticConfig::instance();
+    auto vdev_context = r_cast< const hs_vdev_context* >(vinfo.get_user_private());
 
-    /* create blkstore */
-    blkstore_blob* blob = r_cast< blkstore_blob* >(vb->context_data);
-
-    switch (blob->type) {
-    case blkstore_type::DATA_LOGDEV_STORE:
-        if (has_log_service() && m_data_log_store_size_pct) {
-            m_log_service->open_vdev(vb, LogStoreService::DATA_LOG_FAMILY_IDX);
+    switch (vdev_context->type) {
+    case hs_vdev_type_t::DATA_LOGDEV_VDEV:
+        if (has_log_service()) {
+            ret_vdev = m_log_service->open_vdev(vinfo, LogStoreService::DATA_LOG_FAMILY_IDX, load_existing);
         }
         break;
-    case blkstore_type::CTRL_LOGDEV_STORE:
-        if (has_log_service() && m_ctrl_log_store_size_pct) {
-            m_log_service->open_vdev(vb, LogStoreService::CTRL_LOG_FAMILY_IDX);
+
+    case hs_vdev_type_t::CTRL_LOGDEV_VDEV:
+        if (has_log_service()) {
+            ret_vdev = m_log_service->open_vdev(vinfo, LogStoreService::CTRL_LOG_FAMILY_IDX, load_existing);
         }
         break;
-    case blkstore_type::META_STORE:
-        if (has_meta_service()) { m_meta_service->open_vdev(vb); }
+
+    case hs_vdev_type_t::META_VDEV:
+        if (has_meta_service()) { ret_vdev = m_meta_service->open_vdev(vinfo, load_existing); }
         break;
 
-    case blkstore_type::INDEX_STORE:
-        if (has_index_service()) { m_index_service->open_vdev(vb); }
+    case hs_vdev_type_t::INDEX_VDEV:
+        if (has_index_service()) { ret_vdev = m_index_service->open_vdev(vinfo, load_existing); }
         break;
 
-    case blkstore_type::DATA_STORE:
-        if (has_data_service()) { m_data_service->open_vdev(vb); }
+    case hs_vdev_type_t::DATA_VDEV:
+        if (has_data_service()) { ret_vdev = m_data_service->open_vdev(vinfo, load_existing); }
         break;
 
     default:
-        HS_LOG_ASSERT(0, "Unknown blkstore_type {}", blob->type);
+        HS_LOG_ASSERT(0, "Unknown vdev_type {}", vdev_context->type);
     }
+
+    return ret_vdev;
 }
 
-uint64_t HomeStore::pct_to_size(const float pct, const PhysicalDevGroup pdev_group) const {
-    uint64_t sz = uint64_cast((pct * static_cast< double >(m_dev_mgr->total_cap(pdev_group))) / 100);
-    return sisl::round_up(sz, m_dev_mgr->phys_page_size(pdev_group));
+uint64_t HomeStore::pct_to_size(float pct, HSDevType dev_type) const {
+    uint64_t sz = uint64_cast((pct * static_cast< double >(m_dev_mgr->total_capacity())) / 100);
```

Comment:
No, we want the percentage to be specified on overall size than device type size. Otherwise consumer of homestore will have very difficult specifying the percentage of each service during formatting

Reply:
ok, make sense